src/components/tables/DataTable/DataTableLazy.stories.tsx

```diff
@@ -0,0 +1,173 @@
+/* Copyright (c) Fortanix, Inc.
+|* This Source Code Form is subject to the terms of the Mozilla Public License, v. 2.0. If a copy of
+|* the MPL was not distributed with this file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import React, { useState, useMemo, useCallback } from 'react';
+
+import { delay } from '../util/async_util.ts';
+import { type User, generateData } from '../util/generateData.ts';
+import { sortDateTime } from '../util/sorting_util.ts';
+
+import { Button } from '../../actions/Button/Button.tsx';
+import { Panel } from '../../containers/Panel/Panel.tsx';
+import * as DataTableLazy from './DataTableLazy.tsx';
+
+export default {
+  component: DataTableLazy.DataTableLazy,
+  tags: ['autodocs'],
+  parameters: {
+    layout: 'centered',
+  },
+};
+
+type dataTeableLazyTemplateProps = DataTableLazy.TableProviderLazyProps<User> & { delay: number, items: Array<User> };
+const DataTableLazyTemplate = (props: dataTeableLazyTemplateProps) => {
+  const columns = useMemo(() => props.columns, [props.columns]);
+  const items = useMemo(() => props.items, [props.items]);
+  const delayQuery = props.delay ?? null;
+
+  const [itemsProcessed, setItemsProcessed] = useState<DataTableLazy.DataTableQueryResult<User>>({ total: 0, itemsPage: [] });
+
+  const query: DataTableLazy.DataTableQuery<User> = useCallback(
+    async ({ pageIndex, pageSize }) => {
+      if (delayQuery === Number.POSITIVE_INFINITY) return new Promise(() => {}); // Infinite delay
+      if (delayQuery === -1) throw new Error('Failed'); // Simulate failure
+
+      if (delayQuery) await delay(delayQuery);
+
+      // Simulate failure on page 4
+      if (typeof delayQuery === 'number' && delayQuery > 0 && pageIndex + 1 === 4) {
+        throw new Error('Failed');
+      }
+
+      const itemsProcessedPage = items.slice(pageIndex * pageSize, pageIndex * pageSize + pageSize);
+
+      return { total: items.length, itemsPage: itemsProcessedPage };
+    },
+    [items, delayQuery]
+  );
+
+  return (
+    <Panel>
+      <DataTableLazy.TableProviderLazy
+        {...props}
+        columns={columns}
+        items={itemsProcessed}
+        updateItems={setItemsProcessed}
+        query={query}
+      >
+        <DataTableLazy.Search />
+        <DataTableLazy.DataTableLazy
+          placeholderEmpty={
+            <DataTableLazy.DataTablePlaceholderEmpty
+              icon="file"
+              title="No users"
+              actions={
+                <DataTableLazy.PlaceholderEmptyAction>
+                  <Button variant="secondary" onClick={() => {}}>Action 1</Button>
```

Comment:
Can you change the `onClick` here to `onPress`? Also for any other usages of `Button`.